### PR TITLE
Generate reports even on run failures

### PR DESCRIPTION
**Pull request by waxtell**
Hello Team,

I hope this PR finds you well!

I love how the CLI now returns an error code for failed runs, however, we would still very much like to generate report artifacts even when the run contains one or more failures.

Thank you, in advance for your time and consideration!

Best Regards,
Will

### CODE REVIEW

File: bin/run_functions.js [hunk 1]
```diff
@@ -496,6 +493,10 @@ function executeProject(filename, project, options) {
                         }
                     }
                     jobId = null;
+
```
Comment (waxtell) [inline bin/run_functions.js:496]: Relocating the fail/exit logic here ensures that:

1. Reports are still run even for failed runs and
2. jobId is set to null which avoids the "To cancel the job started, please use..." message

File: bin/jobs_functions.js [hunk 2]
```diff
@@ -231,7 +231,6 @@ function reportForTestJob(testjobId, outputFolder, fileName, format) {
                     util.printErrorAndExit(`Status code: ${response.status}`);
                 } else {
                     util.output('Report created successfully');
-                    process.exit(0);
```
Comment (waxtell) [inline bin/jobs_functions.js:234]: Exiting the process as such ostensibly "cancels" the stream write operation resulting in 0 length report files.

**Comment by lootic** [comment]
Hi! Thank you for this PR. We did notice our mistake and did make our own PR to fix it here:
https://github.com/SmartBear/testengine-cli/pull/30

We hope it will reach you soon.

**Comment by waxtell** [comment]
Fantastic!  Thank you very much!

**Pull request closed without merging**